submissions/7SergSmith7/Tiny JS World/index.js

```diff
@@ -0,0 +1,105 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+class Habitant {
+  constructor(species, name, gender, phrase, friends = [], legs, props = []) {
+    this.species = species;
+    this.legs = legs;
+    this.name = name;
+    this.gender = gender;
+    this.phrase = phrase;
+    this.friends = friends;
+    this.addFriends(friends);
+    this.props = props;
+    this.addProps();
+  }
+  addProps() {
+    this.props = ["species", "legs", "name", "gender", "phrase"];
+  }
+  addFriends(friends) {
+    this.friends = [...this.friends, ...friends];
+  }
+
+  listFriends() {
+    return this.friends.length
+      ? this.friends.map((friend) => friend.name)
+      : "I am loner";
+  }
+
+  sayPhrase() {
+    return this.phrase;
+  }
+
+  toString() {
+    this.props = this.props.map((prop) => `<strong>${this[prop]}</strong>`);
+
+    return `Hi! I am ${this.props[0]}, my name is ${
+      this.props[2]
+    }, my gender is ${this.props[3]}.
+    My phrase: ${
+      this.props[4]
+    } Friends: <strong>${this.listFriends()}</strong>. I have ${
+      this.props[1]
+    } legs `;
```

Comment:
Referring to the props by indices looks cryptic. Imagine, the class has dozens of props. Easy to mess up with indices.

Deconstruction at our rescue
Hint: ``[var1, var2, var3] = ['var1', 'var2', 'var3'].map(varName => `<strong>${this[varName]}</strong>`) ``

Reply:
Please look at this option